docs/pages/cairo/enum.md

```diff
@@ -1,21 +1,69 @@
 ## Enum
 
-Enums are very useful in game design, as they simplify the creation of clean, complex logic.
+## What is Enum
+Enums, short for "enumerations," are a way to define a custom data type that consists of a fixed set of named values, called variants. Enums are useful for representing a collection of related values where each value is distinct and has a specific meaning. Enums are particularly useful in game development for representing game states, player actions, or any other set of related constants that a game might need to track.
+```rust
+# [derive(Serde, Copy, Drop, Introspect, PartialEq, Print)]
```

Comment:
Perhaps we may not want to have all those derive here, while only introducing the concept?

Reply:
I would rather add a comment or reference to some Cairo documentation about those attributes to not have them in a simple example, wdyt?

Reply:
`Introspect` is necessary if the enum must be into a model.
`Serde` is necessary if the enum is used as input/output of an external function of a dojo contract.
`Copy`, `Drop`, `PartialEq`, `Print` may vary depend on the use-case.